pkg/kademlia/routing_helpers.go

```diff
@@ -234,27 +223,16 @@ func (rt *RoutingTable) getNodeIDsWithinKBucket(bID bucketID) (storj.NodeIDList,
 	}
 	left := endpoints[0]
 	right := endpoints[1]
-	var nodeIDsBytes [][]byte
-	allNodeIDsBytes, err := rt.nodeBucketDB.List(nil, 0)
-	if err != nil {
-		return nil, RoutingErr.New("could not list nodes %s", err)
-	}
-	for _, v := range allNodeIDsBytes {
-		if (bytes.Compare(v, left[:]) > 0) && (bytes.Compare(v, right[:]) <= 0) {
-			nodeIDsBytes = append(nodeIDsBytes, v)
-			if len(nodeIDsBytes) == rt.bucketSize {
-				break
-			}
+	var ids []storj.NodeID
+
+	err = rt.iterateNodes(storj.NodeID{}, func(nodeID storj.NodeID, protoNode []byte) error {
```

Comment:
is there a reason we are not supplying `left` as the starting point? Also doing iteration over the nodes twice seems wasteful.

Reply:
that's fair, I'll condense it